EIPS/eip-8011.md

```diff
@@ -0,0 +1,172 @@
+
+After executing the ith EVM operation of a transaction, say `OP_i`, with a cost vector `gas_cost_vector_i = (compute_cost_i, access_cost_i, size_cost_i, memory_cost_i, state_cost_i, history_cost_i)`, the following variables are updated:
+
+- Remaining available gas for the transaction:
+  - `gas = gas - sum(compute_cost_i, access_cost_i, size_cost_i, memory_cost_i, state_cost_i, history_cost_i)`
+- Total used gas by the transaction so far:
+  - `gas_used = gas_used + sum(compute_cost_i, access_cost_i, size_cost_i, memory_cost_i, state_cost_i, history_cost_i)`
+- Used gas vector by the transaction so far:
+  - `gas_used_vector = (compute_gas_used + compute_cost_i, access_gas_used + access_cost_i, size_gas_used + size_cost_i, memory_gas_used + memory_cost_i, state_gas_used + state_cost_i, history_gas_used + history_cost_i)`
+
+At the transaction level, everything remains the same. A transaction still has the same one-dimensional `gas_limit`. The transaction's out-of-gas condition is still defined as `gas_used <= gas_limit`. Additionally, the transaction's fee is also computed with `gas_used`.
+
+The key change is the introduction of `gas_used_vector` as a new variable that tracks the gas used by each resource at the transaction level. This variable is returned by the `execute_transaction` function, in addition to `gas_used`:
+
+```python
+def execute_transaction(self, transaction: NormalizedTransaction, effective_gas_price: int) -> int, narray: pass
+```
+
+### Block header extension
+
+The current header encoding is extended with a new 64-bit unsigned integer field named `max_gas_metered`. This integer corresponds to the total gas units used by the bottleneck resource, i.e., the resource with the largest used gas in the block. This variable is computed as follows:
+
+```python
+def compute_block_max_gas_metered(block: Block) -> int:
+    transactions = self.transactions(block)
+    block_gas_used_vector = array(0, 0, 0, 0, 0, 0)
+    for transaction in transactions:
+        gas_used, gas_used_vector = self.execute_transaction(transaction, effective_gas_price)
+        block_gas_used_vector += gas_used_vector
+    max_gas_metered = max(block_gas_used_vector)
+    return max_gas_metered
+```
+
+The header sequence with the new field is `[..., parent_beacon_block_root, requests_hash, max_gas_metered]`.
+
+### Block validity condition
+
+The block validity conditions are modified to replace `block.gas_used` with `block.max_gas_metered`:
+
+```python
+assert block.max_gas_metered <= block.gas_limit, 'invalid block: too much gas used'
+```
+
+### Base fee update rule
+
+The block validity conditions are modified to replace `parent.gas_used` with `parent.max_gas_metered`:
+
+```python
+gas_used_delta = parent.max_gas_metered - parent.gas_target
```

Comment:
this is also a massive re-interpretation of the gas target and should be mentioned upfront in the abstract